tasks/image-segmentation-yolov8/src/validate.py

```diff
@@ -0,0 +1,213 @@
+from pathlib import Path
+
+import json
+import csv
+import logging
+
+from ultralytics import YOLO
+from PIL import Image
+from pycocotools import mask as maskCoco
+from matplotlib import pyplot as plt
+from coretex import TaskRun, ImageDataset, folder_manager, Model, ImageSample, ImageDatasetClasses
+
+import numpy as np
+
+from .dataset import prepareDataset, createYamlFile
+
+
+def fetchCsvDatasetAcc(csvSamplesPath: Path) -> list[float]:
+    with csvSamplesPath.open("r") as file:
+        reader = csv.DictReader(file)
+        csvDatasetAcc = [float(row["accuracy"]) for row in reader]
+
+    return csvDatasetAcc
+
+
+def fetchCsvClassesData(labels: list[str], csvSamplesPath: Path) -> dict[str, list[float]]:
+    csvClassesData: dict[str, list[float]] = {}
+
+    with csvSamplesPath.open("r") as file:
+        reader = csv.DictReader(file)
+        for label in labels:
+            classAcc = [float(row[label]) for row in reader]
+            csvClassesData[label] = classAcc
+
+    return csvClassesData
+
+
+def generateFieldNamesDataset(labels: list[str]) -> list[str]:
+    fieldNamesDataset: list[str] = []
+
+    for label in labels:
+        fieldNamesDataset.append(f"{label} acc")
+        fieldNamesDataset.append(f'{label} acc STD')
+
+    fieldNamesDataset.extend(["accuracy", "accuracy STD"])
+
+    return fieldNamesDataset
+
+
+def createCsvDatasetResults(taskRun: TaskRun[ImageDataset], csvSamplesPath: Path) -> float:
+    fieldNamesDataset = generateFieldNamesDataset(taskRun.dataset.classes.labels)
+
+    csvDatasetPath = folder_manager.createTempFolder("csv_dataset") / "dataset_results.csv"
+
+    csvClassesData = fetchCsvClassesData(taskRun.dataset.classes.labels, csvSamplesPath)
+    csvDatasetAcc = fetchCsvDatasetAcc(csvSamplesPath)
+
+    csvDatasetResults: dict[str, str] = {}
+    for label in taskRun.dataset.classes.labels:
+        csvDatasetResults[f"{label} acc"] = f"{np.mean(csvClassesData[label]):.2f}"
```

Comment:
Use `Accuracy` and `Accuracy stdev` as column names everywhere to keep it consistent.